core/chains/evm/client/errors.go

```diff
@@ -227,7 +228,11 @@ var zkSync = ClientErrors{
 	Fatal: regexp.MustCompile(`(?:: |^)(?:exceeds block gas limit|intrinsic gas too low|Not enough gas for transaction validation|Failed to pay the fee to the operator|Error function_selector = 0x, data = 0x|invalid sender. can't start a transaction from a non-account|max(?: priority)? fee per (?:gas|pubdata byte) higher than 2\^64-1|oversized data. max: \d+; actual: \d+)$`),
 }
 
-var clients = []ClientErrors{parity, geth, arbitrum, metis, substrate, avalanche, nethermind, harmony, besu, erigon, klaytn, celo, zkSync}
+var zkEvm = ClientErrors{
+	OutOfCounters: regexp.MustCompile(`(?:: |^)not enough keccak counters to continue the execution$`),
```

Comment:
This isn't the only error they return when out of counters, it depends on why they're being exceeded. There are currently eight ([source](https://github.com/0xPolygonHermez/zkevm-node/blob/aae30e9c79bdf363814e7fe2a3df9b34e855c998/state/runtime/runtime.go#L28-L43)):
```
not enough step counters to continue the execution
not enough keccak counters to continue the execution
not enough binary counters to continue the execution
not enough memory align counters to continue the execution
not enough arith counters to continue the execution
not enough padding counters to continue the execution
not enough poseidon counters to continue the execution
not enough sha256 counters to continue the execution
```

We could match them all using:
```suggestion
	OutOfCounters: regexp.MustCompile(`(?:: |^)not enough .* counters to continue the execution$`),
```

Reply:
Ah didn't know that, thanks for calling it out. I've updated the regex accordingly